x/app-router/keeper/liquidstake.go

```diff
@@ -0,0 +1,84 @@
+package keeper
+
+import (
+	"github.com/armon/go-metrics"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	transfertypes "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types"
+	channeltypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	ibcexported "github.com/cosmos/ibc-go/v3/modules/core/exported"
+
+	"github.com/Stride-Labs/stride/v3/x/app-router/types"
+	stakeibckeeper "github.com/Stride-Labs/stride/v3/x/stakeibc/keeper"
+	stakeibctypes "github.com/Stride-Labs/stride/v3/x/stakeibc/types"
+)
+
+func (k Keeper) TryLiquidStaking(
+	ctx sdk.Context,
+	packet channeltypes.Packet,
+	newData transfertypes.FungibleTokenPacketData,
+	parsedReceiver *types.ParsedReceiver,
+	ack ibcexported.Acknowledgement,
+) ibcexported.Acknowledgement {
+	params := k.GetParams(ctx)
+	if !params.Active {
+		return channeltypes.NewErrorAcknowledgement("packet forwarding param is not active")
+	}
+	// recalculate denom, skip checks that were already done in app.OnRecvPacket
+	var denom string
+	// In this case, we can't process a liquid staking transaction, because we're dealing with STRD tokens
+	if transfertypes.ReceiverChainIsSource(packet.GetSourcePort(), packet.GetSourceChannel(), newData.Denom) {
+		return channeltypes.NewErrorAcknowledgement("not a supported token for liquid staking")
+	} else {
+		prefixedDenom := transfertypes.GetDenomPrefix(packet.GetDestPort(), packet.GetDestChannel()) + newData.Denom
```

Comment:
Can you add a comment here explaining this line of code?

Reply:
This code is removed since it was converting base denom to ibc denom and back to base denom. https://github.com/Stride-Labs/stride/pull/432/commits/d17ffb6028a32eba576bf75d10c226dfc539278e